pkgs/build-support/libredirect/default.nix

```diff
@@ -15,8 +15,17 @@ stdenv.mkDerivation rec {
 
   buildPhase = ''
     runHook preBuild
-
-    $CC -Wall -std=c99 -O3 -fPIC -ldl -shared \
+    ${if stdenv.isDarwin then ''
+    # use the system clang, and break the sandbox here. Why? We will use this library to inject
+    # into dyld.  We don't know on which macOS we are running, and while we might know on which
+    # architecture we are running, it's just easer to create a fat library that contains all
+    # possible architecture slices.  Our llvmPackages clang is not multi-target, hence the system
+    # clang.  :'(
+    /usr/bin/clang -arch x86_64 -arch arm64 -arch arm64e \
```

Comment:
Can't you use lipo to make a fat binary?

Reply:
That's what `clang` will call out if you provide multiple `-arch`.

Reply:
If I'm not mistaken, that's only true for Apple Clang which is impure and not installed on Hydra.